lib/puppet-lint/plugins/check_classes.rb

```diff
@@ -55,19 +55,19 @@ def check
 PuppetLint.new_check(:names_containing_dash) do
   def check
     (class_indexes + defined_type_indexes).each do |class_idx|
-      if class_idx[:name_token].value.include? '-'
-        if class_idx[:type] == :CLASS
-          obj_type = 'class'
-        else
-          obj_type = 'defined type'
-        end
+      return unless class_idx[:name_token].value.include? '-'
```

Comment:
This should be `next unless`, otherwise the whole check function will return early rather than just skipping that element of the array.